### PR TITLE
CORE-564: CLI should manage odd directory names more intelligently

### PR DESCRIPTION
**Pull request by mattaudesse**
Notes:

It might be easiest to glance at `README.md` before eyeballing the other files.

b57e025 is an unrelated change but I wanted to sneak it in anyway lest I forget about it... Without that change `stack` could fail to build but the script would keep rolling.

---

Our current CLI will fail when using a directory with spaces or wacky characters, e.g.:
```sh
$ mkdir -p "  foo _ bar @ bazz  "
```

These changes are intended to handle both the UNIXy-path-handling glitches (quoting) and Docker container names (replace non-alphanumerics with single "-" characters + other formatting rules: see `appProj` definition).

### CODE REVIEW

**Review by jeapostrophe — APPROVED** [review]

File: examples/cli-tricky-directories/A directory name with spaces /CNAME [hunk 1]
```diff
@@ -0,0 +1 @@
+a-directory-name-with-spaces
```
Comment (jeapostrophe) [inline examples/cli-tricky-directories/A directory name with spaces /CNAME:1]: I was really confused why we were making DNS entries for these until I read the rest of the code :)

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: examples/cli-tricky-directories/«Інтернет»/лептоп/або смартфон/CNAME [hunk 2]
```diff
@@ -0,0 +1 @@
+x
```
[the remark this answers is not in the document]
Comment (mattaudesse) [reply examples/cli-tricky-directories/«Інтернет»/лептоп/або смартфон/CNAME:1]: Nope; containers also get the process ID (which is what `1234` represents in the README).

For `package.json` I figured `"name": "x"` is lame but anyone doing serious work on their project will be tweaking their config anyway to add more dependencies, `npm run` hooks, etc.

I'm also thinking about cloning `git` repos where `package.json` lives in the root - nothing guarantees the directory name will match others' or make sense.

I agree this solution is disappointing though... Want me to pretty it up some more?

File: examples/cli-tricky-directories/A directory name with spaces /CNAME (hunk 1 above)
Comment (mattaudesse) [reply examples/cli-tricky-directories/A directory name with spaces /CNAME]: CNAME was a BADNAME.

File: examples/cli-tricky-directories/What' about 'this'?/index.mjs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import {loadStdlib} from '@reach-sh/stdlib';
```
[the remark this answers is not in the document]
Comment (mattaudesse) [reply examples/cli-tricky-directories/What' about 'this'?/index.mjs]: I've pruned a bunch of cruft in 3daa4dd18b165962e9684ed1b0f1b60a9b8ecf4d.

File: examples/cli-tricky-directories/«Інтернет»/лептоп/або смартфон/CNAME (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeapostrophe) [reply examples/cli-tricky-directories/«Інтернет»/лептоп/або смартфон/CNAME]: I understand. Thanks.

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**